### PR TITLE
Possibly improve tab loading performance and add better logging

### PR DESCRIPTION
**Pull request by Standard8**
Whilst looking for other information, I discovered we can probably improve the loading performance per-tab and avoid the overhead of loading the large objects for each tab - doing it once per process:

https://developer.mozilla.org/en-US/Firefox/Multiprocess_Firefox/Message_Manager/Performance#Declaring_stateless_functions_once_per_process

Additionally, I'm switching to Log.jsm, for better logging - especially different levels. We might take some of it out eventually, but I also think it is useful to leave some of it in, and have at least two different levels. Log.jsm seems to make this easier.

### CODE REVIEW

File: add-on/content/SerpMonitor.jsm [hunk 1]
```diff
@@ -125,19 +131,20 @@ this.SerpMonitor = {
...
       if (resultDomainInfo &&
         uri.filePath.substring(1).startsWith(resultDomainInfo.prefix) &&
                  this.serpTabs.has(triggerURI.spec)) {
         let info = this.serpTabs.get(triggerURI.spec);
-        Cu.reportError(`${resultDomainInfo.sap} ad click`); // XXX
 
         let histogram = Services.telemetry.getKeyedHistogramById("SEARCH_COUNTS");
+        log.info(`Reporting to Telemetry: ${info.sap}.adclick:unknown:${info.code}`);
         histogram.add(`${info.sap}.adclick:unknown:${info.code}`);
-        this.serpTabs.delete(triggerURI.spec);
```
Comment (past) [inline add-on/content/SerpMonitor.jsm]: This change is unrelated, right? It ensures ad clicks are counted once, even though a single GET request ends up triggering the activity observer multiple times.

**Review by past — APPROVED** [review]
Apart from the seemingly unrelated change, the rest looks good to me.

File: add-on/content/SerpMonitor.jsm (hunk 1 above)
Comment (Standard8) [reply add-on/content/SerpMonitor.jsm]: Thanks, I've added that back in for now.

I also fixed the message reporting in bootstrap.js to include the message name as well as the data.
